src/coreclr/gc/gc.cpp

```diff
@@ -2248,12 +2248,26 @@ uint64_t    gc_heap::gc_last_ephemeral_decommit_time = 0;
 
 CLRCriticalSection gc_heap::check_commit_cs;
 
+CLRCriticalSection gc_heap::decommit_lock;
+
 size_t      gc_heap::current_total_committed = 0;
 
 size_t      gc_heap::committed_by_oh[recorded_committed_bucket_counts];
 
 size_t      gc_heap::current_total_committed_bookkeeping = 0;
 
+#ifdef COMMITTED_BYTES_SHADOW
+
+CLRCriticalSection gc_heap::check_commit_shadow_cs;
```

Comment:
it seems like you are adding these new fields but we already have fields in debug build that we use when hardllimit is on. we could use the same fields to always record when this define is defined.